### ISSUE TITLE
Add some documentation for NFS

### ISSUE THREAD

**Issue by willingc** [jupyterhub/jupyterhub#839]
From discussion on the gitter channel, it would be great to add a guide or tips for using NFS and file persistence. Please leave your suggestions in as much detail as possible in this issue. Thanks!

From gitter:

```
 Martin Angelov @mangelov95 03:30
Hello, is there a guide somewhere how to configure NFS mounting of home directories of the users, so that their files are persistent and saved when the Notebook docker container is removed? I am using jupyterhub with Docker swarm and SystemUserSpawner.
I didn't manage to find any good information about using NFS with jupyterhub and at the moment when I'm trying to do it myself I get Spawner failed to start [status=ExitCode=1, Error='', FinishedAt=2016-10-28T10:27:36.240814672Z] with no further explanation of the error in the jupyterhub logs

Justin Tyberg @jtyberg 05:28
@mangelov95 I've had success using Convoy Docker volume plugin for Docker/swarm/NFS. Just run the convoy daemon on each swarm node. The jupyterhub config is pretty straightforward - just mount the Docker volumes by name (one for each user) and set the volume driver to convoy.
# Mount the real user's Docker volume on the host to the notebook user's
# notebook directory in the container
c.DockerSpawner.volumes = { 'jupyterhub-user-{username}': notebook_dir }
c.DockerSpawner.extra_create_kwargs.update({ 'volume_driver': 'convoy' })

Joshua Milas @DeepHorizons 06:20
I'm using NFS, but I mounted it on the server and then add the volume like a regular directory.

Carol Willing @willingc 08:16
@jtyberg Would you mind opening an issue with some suggested steps using Convoy and NFS so we can add the suggested config under the troubleshooting section of the docs. Thanks!

Martin Angelov @mangelov95 12:32
@willingc @jtyberg This is a great idea! Although I solved my issue by mounting the NFS directory to the Docker host VM and then modifying the host_homedir_format_stringoption, some more documentation about NFS would be awesome and appreciated.

Yuvi Panda @yuvipanda 12:35
@mangelov95 that's what we do with kubernetes too btw 
```

**Comment by mangelov95** [comment]
I managed to get NFS mount working when using Jupyterhub with Docker Swarm and `dockerspawner.SystemUserSpawner` by:

- Mounting the NFS directory on the host virtual machine, on which was the Docker host
- Setting the `SystemUserSpawner.host_homedir_format_string` option in `jupyterhub_config.py` to point to the mounted directory.  That way when the notebook image is started it will initialise and use the NFS directory.

**Comment by barrachri** [comment]
Any experience with GlusterFS of Ceph?

**Comment by willingc** [comment]
@minrk I'm going to bump this to 0.9 and include as a sprint item for JupyterCon.

**Comment by minrk** [comment]
Great!

**Comment by willingc** [comment]
Closed by PR #1960

**Comment by ltalirz** [comment]
@willingc I'm new to jupyterhub, so I might overlook something, but I fail to see how #1960 addresses this issue. Did you mean to mention another PR?
